app/assets/javascripts/discourse/app/components/composer-fullscreen-prompt.js

```diff
@@ -1,27 +1,21 @@
+import { action } from "@ember/object";
 import GlimmerComponent from "@glimmer/component";
 import I18n from "I18n";
-import { next } from "@ember/runloop";
 
 export default class ComposerFullscreenPrompt extends GlimmerComponent {
-  constructor() {
-    super(...arguments);
-    this.#setupFullscreenPrompt();
+  @action
+  registerAnimationListener(element) {
+    this.#addAnimationEventListener(element);
   }
 
-  #setupFullscreenPrompt() {
-    next(() => {
-      const promptElement = document.querySelector(
-        ".composer-fullscreen-prompt"
-      );
-
-      promptElement?.addEventListener(
-        "animationend",
-        () => {
-          this.args.removeFullScreenExitPrompt();
-        },
-        { once: true }
-      );
-    });
+  #addAnimationEventListener(element) {
+    element?.addEventListener(
```

Comment:
The element will always be present so there is no need for ? anymore.

Reply:
You're right. Sorry, just a habit :smile: